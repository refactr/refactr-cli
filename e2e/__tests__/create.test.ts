import { executeWithStdoutOnly } from '../helpers/execute';
import { loadFixtures } from './fixtures';

beforeAll(async () => {
  return await loadFixtures();
});

describe('refactrctl create', () => {
  test('throws on missing subcommand', async () => {
    await expect(executeWithStdoutOnly(['create'])).rejects.toMatchSnapshot();
  });

  describe('credential', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['create', 'credential'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['create', 'pipeline'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('pipeline-revision', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['create', 'pipeline-revision'])
      ).rejects.toMatchSnapshot();
    });
  });

  describe('project', () => {
    test('throws on missing arguments', async () => {
      await expect(
        executeWithStdoutOnly(['create', 'project'])
      ).rejects.toMatchSnapshot();
    });
  });
});
