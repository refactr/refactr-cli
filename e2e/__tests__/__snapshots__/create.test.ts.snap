// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`refactrctl create credential throws on missing arguments 1`] = `
"refactrctl create credential

Create a new credential

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this credential will be created under  [string] [required]
      --name        Name of the project  [string] [required]
      --data        Credential data. Must be valid JSON value.  [string] [required]
      --id          Credential reference id  [required]
      --type        Credential type  [required] [choices: \\"api_token\\", \\"aws_access_key\\", \\"azure_service_principal\\", \\"bearer_token\\", \\"generic\\", \\"ssh_key\\", \\"username_password\\", \\"vault_app_role\\"]

Missing required arguments: project-id, name, data, id, type
"
`;

exports[`refactrctl create pipeline throws on missing arguments 1`] = `
"refactrctl create pipeline [options]

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]
      --project-id  Project this pipeline belongs to  [string] [required]
      --name        Name of the pipeline  [string] [required]

Missing required arguments: project-id, name
"
`;

exports[`refactrctl create pipeline-revision throws on missing arguments 1`] = `
"refactrctl create pipeline-revision <input>

Positionals:
  input  [string] [required]

Options:
      --version      Show version number  [boolean]
  -v, --verbose      Print detailed output  [boolean]
      --format       Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address      Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token   Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help         Show help  [boolean]
      --project-id   Project this pipeline belongs to  [string] [required]
      --pipeline-id  Pipeline  [required]

Not enough non-option arguments: got 0, need at least 1
"
`;

exports[`refactrctl create project throws on missing arguments 1`] = `
"refactrctl create project

Options:
      --version          Show version number  [boolean]
  -v, --verbose          Print detailed output  [boolean]
      --format           Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address          Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token       Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help             Show help  [boolean]
      --organization-id  Project this pipeline belongs to  [string] [required]
      --name             Name of the project  [string] [required]

Missing required arguments: organization-id, name
"
`;

exports[`refactrctl create throws on missing subcommand 1`] = `
"refactrctl create

Commands:
  refactrctl create project
  refactrctl create pipeline-revision <input>
  refactrctl create pipeline [options]
  refactrctl create credential                 Create a new credential
  refactrctl create job                        Create a new job

Options:
      --version     Show version number  [boolean]
  -v, --verbose     Print detailed output  [boolean]
      --format      Specifies output format  [choices: \\"wide\\", \\"json\\", \\"yaml\\"] [default: \\"wide\\"]
      --address     Address of the Refactr API server. This can also be specified via the REFACTR_ADDRESS environment variable. Defaults to https://api.refactr.it/v1  [string]
      --auth-token  Authentication token. This can also be specified via the REFACTR_AUTH_TOKEN environment variable  [string]
      --help        Show help  [boolean]

Command must be specified.
"
`;
